api.planx.uk/inviteToPay/paymentRequest.ts

```diff
@@ -83,17 +91,30 @@ export async function buildPaymentPayload(
   if (!req.query.returnURL) {
     return next(
       new ServerError({
-        message: "missing required returnURL query param",
+        message: "Missing required returnURL query param",
+        status: 400,
+      }),
+    );
+  }
+
+  if (!req.query.sessionId) {
+    return next(
+      new ServerError({
+        message: "Missing required sessionId query param",
         status: 400,
       }),
     );
   }
-  req.body = {
-    amount: req.params.paymentAmount,
-    reference: req.query.sessionId,
+
+  const createPaymentBody: GovPayCreatePayment = {
+    amount: parseInt(req.params.paymentAmount),
```

Comment:
can you remind me how `req.params.paymentAmount` relates to `payment_requests.payment_amount` here - are they the both formatted as pence? if so, does `parseInt()` actually work as expected here (eg `20600` to £206)? 

sorry if this is something obvious i'm missing!

Reply:
No worries! I'm actually really keen to come back to all this and do some typing / documentation to tidy up here as I've found the data flow of the PaymentRequest middlewares a little hard to follow at times.

GovPay always handles payment amount in pence, and this is also what we store in the DB (as a number).

The `fetchPaymentRequestDetails()` middleware gets these details from the db when the API endpoints are called. It gets `payment_requests.payment_amount` as a number (in pence). To pass this onto the next middleware layer it was adding this as a query param. As it was untyped TS wasn't flagging it as a problem that we were going -

`payment_requests.payment_amount` → `req.params.paymentAmount` → GovPay API `payment_amount`

Which should be - 

number → string → number

So now we're just converting the types correctly.

Honestly a better solution here (but a wider refactor) would be to use something like `res.locals` docs (https://expressjs.com/en/5x/api.html#res.locals) to pass these values between middleware layers.

If it seems worthwhile (and makes things easier to follow) I'd be happy to pick this up just now tbh!

Reply:
Thanks for clarification - prod `payment_requests` table has an outdated mix of `103` and `20600` from early testing (and no "real" requests since) which always confuses me a bit!

Reply:
Yeah I think this was a bug which we had in place during development of the feature which is now resolved, there's a PR somewhere where I added this comment on the column - 

<img width="280" alt="image" src="https://github.com/theopensystemslab/planx-new/assets/20502206/15a1d365-e4a8-46eb-83ee-57a3d8f59724">